### PR TITLE
[synthetics] Add `locations` and `private_locations` lightweight options

### PR DESCRIPTION
**Pull request by colleenmcginnis**
Follow up from https://github.com/elastic/observability-docs/pull/3299#pullrequestreview-1708159244

Adds `locations` and `private_locations` to the lightweight monitor configuration options.

### CODE REVIEW

File: docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc [hunk 1]
```diff
@@ -234,6 +234,62 @@ By default, monitors are automatically retested if the monitor goes from "up" to
...
+* Run the <<elastic-synthetics-locations-command,`elastic-synthetics locations` command>>.
+* Go to *Synthetics* -> *Management* and click *Create monitor*. Locations will be listed in _Locations_.
+
+// If no `locations` or `private_locations` are specified in the monitor configuration, but is defined in the <<synthetics-configuration,project configuration file>>, the monitors will be pushed to the locations listed in the project configuration file.
+
+*Examples*:
+
+[source,yaml]
+----
+locations: ["asia-northeast1-a", "asia-south1-a"]
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc:255]: This would be similar to the public location list we have declared here - https://github.com/elastic/synthetics/blob/d787cff0232536152a918f7964ab4e3a49977c50/src/locations/public-locations.ts#L27-L38

Internally we map these to these long ids to make it easier for devs to configure locations.

File: docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc [hunk 2]
```diff
@@ -218,6 +218,62 @@ Enable TLS certificate alerts on this monitor.
...
+| [[monitor-private_locations]] *`private_locations`*
+(list of <<synthetics-lightweight-data-string,string>>s)
+a| The <<synthetics-private-location,{private-location}s>> to which the monitors will be deployed. These {private-location}s refer to locations hosted and managed by you, whereas  `locations` are hosted by Elastic. You can specify a {private-location} using the location's name.
+
+To list available {private-location}s you can:
+
+* Run the <<elastic-synthetics-locations-command,`elastic-synthetics locations` command>> with the {kib} URL for the deployment from which to fetch available locations.
+* Go to *Synthetics* -> *Management* and click *Create monitor*. {private-location}s will be listed in _Locations_.
+
+// If no `locations` or `private_locations` are specified in the monitor configuration, but is defined in the <<synthetics-configuration,project configuration file>>, the monitors will be pushed to the locations listed in the project configuration file.
```
[the remark this answers is not in the document]
Comment (vigneshshanmugam) [reply docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc:277]: ++ Same as what happens for the browser monitors. We can even link to those docs instead of having it duplicated here. 

```
This can also be set in the configuration file using
<<synthetics-configuration-monitor,`monitor.privateLocations` in the configuration file>>.
The value defined via the CLI will take precedence.

```

**Review by vigneshshanmugam — DISMISSED** [review]
LGTM

**Comment by colleenmcginnis** [comment]
👋  @elastic/obs-docs this is ready for a code owner review.

File: docs/en/observability/synthetics-configuration.asciidoc [hunk 3]
```diff
@@ -227,4 +227,4 @@ Then the monitor will resume running according to the defined schedule.
 Using `retestOnFailure` can reduce noise related to transient problems.
 // end::monitor-config-options[]
 
-For information on configuring monitors individually, refer to <<synthetics-monitor-use>>.
+For information on configuring monitors individually, for browser monitors refer to <<synthetics-monitor-use>> and for lightweight monitors refer to <<synthetics-lightweight-data-types>>.
```
Comment (dedemorton) [inline docs/en/observability/synthetics-configuration.asciidoc:230]: There's a pile of up "for" clauses here that makes the sentence a little harder to scan...how about something like:
```suggestion
For information on configuring monitors individually, refer to:

* <<synthetics-monitor-use>> for browser monitors 
* <<synthetics-lightweight-data-types>> for lightweight monitors
```

File: docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc [hunk 4]
```diff
@@ -234,6 +234,78 @@ By default, monitors are automatically retested if the monitor goes from "up" to
 retest_on_failure: false
 ----
 
+////////////////////////
+locations
+////////////////////////
+| [[monitor-locations]] *`locations`*
+(list of https://github.com/elastic/synthetics/blob/{synthetics_version}/src/locations/public-locations.ts#L28-L37[`SyntheticsLocationsType`])
+a| Where to deploy the monitor. Monitors can be deployed in multiple locations so that you can detect differences in availability and response times across those locations.
```
Comment (dedemorton) [inline docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc:242]: I'm not hyper aggressive about passive voice, but in this case maybe removing it makes the statement more concise?
```suggestion
a| Where to deploy the monitor. You can deploy monitors in multiple locations to detect differences in availability and response times across those locations.
```

File: docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc [hunk 5]
```diff
@@ -234,6 +234,78 @@ By default, monitors are automatically retested if the monitor goes from "up" to
...
+[source,yaml]
+----
+locations:
+  - japan
+  - india
+----
+
+[NOTE]
+====
+This can also be set in the project configuration file using
```
Comment (dedemorton) [inline docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc:265]: Removes repeated words
```suggestion
This can also be set using
```

File: docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc [hunk 6]
```diff
@@ -234,6 +234,78 @@ By default, monitors are automatically retested if the monitor goes from "up" to
...
+<<synthetics-configuration-monitor,`monitor.locations` in the project configuration file>>
+or via the CLI using the <<elastic-synthetics-push-command,`--location` flag on `push`>>.
+
+The value defined via the CLI will take precedence over the value defined in the lightweight monitor configuration,
+and the value defined in the lightweight monitor configuration will take precedence over the value defined in _project_ configuration file.
```
Comment (dedemorton) [inline docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc:270]:
```suggestion
The value defined via the CLI takes precedence over the value defined in the lightweight monitor configuration,
and the value defined in the lightweight monitor configuration takes precedence over the value defined in _project_ configuration file.
```

File: docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc [hunk 7]
```diff
@@ -234,6 +234,78 @@ By default, monitors are automatically retested if the monitor goes from "up" to
...
+////////////////////////
+private_locations
+////////////////////////
+| [[monitor-private_locations]] *`private_locations`*
+(list of <<synthetics-lightweight-data-string,string>>s)
+a| The <<synthetics-private-location,{private-location}s>> to which the monitors will be deployed. These {private-location}s refer to locations hosted and managed by you, whereas  `locations` are hosted by Elastic. You can specify a {private-location} using the location's name.
+
+To list available {private-location}s you can:
+
+* Run the <<elastic-synthetics-locations-command,`elastic-synthetics locations` command>> with the {kib} URL for the deployment from which to fetch available locations.
```
Comment (dedemorton) [inline docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc:282]: This sentence is a little hard to parse. Would it be accurate to say, "...and specify the {kib} URL of the deployment that has the available locations."

(I don't think "has" is the right word here, but I'm not sure how these locations are retrieved.)

File: docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc [hunk 8]
```diff
@@ -234,6 +234,78 @@ By default, monitors are automatically retested if the monitor goes from "up" to
...
+----
+private_locations:
+  - Private Location 1
+  - Private Location 2
+----
+
+[NOTE]
+====
+This can also be set in the project configuration file using
+<<synthetics-configuration-monitor,`monitor.privateLocations` in the project configuration file>>
```
Comment (dedemorton) [inline docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc:302]:
```suggestion
This can also be set using
<<synthetics-configuration-monitor,`monitor.privateLocations` in the project configuration file>>
```

File: docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc [hunk 9]
```diff
@@ -234,6 +234,78 @@ By default, monitors are automatically retested if the monitor goes from "up" to
...
+or via the CLI using the <<elastic-synthetics-push-command,`--privateLocations` flag on `push`>>.
+
+The value defined via the CLI will take precedence over the value defined in the lightweight monitor configuration,
+and the value defined in the lightweight monitor configuration will take precedence over the value defined in _project_ configuration file.
```
Comment (dedemorton) [inline docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc:306]:
```suggestion
The value defined via the CLI takes precedence over the value defined in the lightweight monitor configuration,
and the value defined in the lightweight monitor configuration takes precedence over the value defined in _project_ configuration file.
```

File: docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc [hunk 10]
```diff
@@ -234,6 +234,78 @@ By default, monitors are automatically retested if the monitor goes from "up" to
 retest_on_failure: false
 ----
 
+////////////////////////
+locations
```
Comment (dedemorton) [inline docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc:238]: General comment that you really can't fix: It's annoying that the user needs to use a scrollbar to view all the content in the table.

**Review by dedemorton — COMMENTED** [review]
Just a few minor edits. Otherwise looks good!

File: docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc (hunk 7 above)
Comment (dedemorton) [reply docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc:282]: If it's a list in Kibana somewhere, you could say, "..and specify the {kib} URL of the deployment that has the list of available locations.

Comment (colleenmcginnis) [reply docs/en/observability/synthetics-reference/lightweight-config/common.asciidoc:282]:
```suggestion
* Run the <<elastic-synthetics-locations-command,`elastic-synthetics locations` command>> and specify the {kib} URL of the deployment. This will fetch all available private locations associated with the deployment.
```

File: docs/en/observability/synthetics-configuration.asciidoc [hunk 11]
```diff
@@ -227,4 +227,7 @@ Then the monitor will resume running according to the defined schedule.
 Using `retestOnFailure` can reduce noise related to transient problems.
 // end::monitor-config-options[]
 
-For information on configuring monitors individually, refer to <<synthetics-monitor-use>>.
+For information on configuring monitors individually, refer to:
+
+* <<synthetics-monitor-use>> for browser monitors 
+* <<synthetics-lightweight-data-types>> for lightweight monitors
```
Comment (paulb-elastic) [inline docs/en/observability/synthetics-configuration.asciidoc:233]: Should this be synthetics-lightweight.html?

**Review by paulb-elastic — APPROVED** [review]
A comment about one of the links, not sure if it's right, but won't block the rest LGTM

File: docs/en/observability/synthetics-configuration.asciidoc (hunk 11 above)
Comment (colleenmcginnis) [reply docs/en/observability/synthetics-configuration.asciidoc:233]: Good catch! 

```suggestion
* <<synthetics-lightweight>> for lightweight monitors
```

**Comment by colleenmcginnis** [comment]
@dedemorton can you take another look and give me a ✅ if it looks ok?

**Review by dedemorton — APPROVED** [review]
LGTM
